typings/index.d.ts

```diff
@@ -1357,11 +1357,11 @@ export class Message<Cached extends boolean = boolean> extends Base {
   public embeds: MessageEmbed[];
   public groupActivityApplication: ClientApplication | null;
   public guildId: Snowflake | null;
-  public readonly guild: Guild | null;
+  public readonly guild: If<Cached, Guild, null>;
   public readonly hasThread: boolean;
   public id: Snowflake;
   public interaction: MessageInteraction | null;
-  public readonly member: GuildMember | null;
+  public readonly member: If<Cached, GuildMember, null>;
```

Comment:
```suggestion
  public readonly member: If<Cached, GuildMember | null>;
```
See @wasdennnoch's comment

Reply:
At that point we might as well keep it as `GuildMember | null`

Reply:
I think this should stay as-is, a message in a guild does not guarantee you a member:
e.g. webhook messages, old messages from someone that left the guild since